paymentsheet/src/main/java/com/stripe/android/paymentelement/confirmation/gpay/GooglePayConfirmationDefinition.kt

```diff
@@ -0,0 +1,179 @@
+package com.stripe.android.paymentelement.confirmation.gpay
+
+import androidx.activity.result.ActivityResultCaller
+import androidx.activity.result.ActivityResultLauncher
+import com.stripe.android.core.strings.resolvableString
+import com.stripe.android.core.utils.UserFacingLogger
+import com.stripe.android.googlepaylauncher.GooglePayEnvironment
+import com.stripe.android.googlepaylauncher.GooglePayPaymentMethodLauncher
+import com.stripe.android.googlepaylauncher.GooglePayPaymentMethodLauncherContractV2
+import com.stripe.android.googlepaylauncher.injection.GooglePayPaymentMethodLauncherFactory
+import com.stripe.android.model.PaymentIntent
+import com.stripe.android.model.SetupIntent
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.paymentelement.confirmation.ConfirmationDefinition
+import com.stripe.android.paymentelement.confirmation.ConfirmationHandler
+import com.stripe.android.paymentelement.confirmation.PaymentMethodConfirmationOption
+import com.stripe.android.paymentelement.confirmation.intent.DeferredIntentConfirmationType
+import com.stripe.android.paymentsheet.PaymentSheet
+import com.stripe.android.paymentsheet.R
+import com.stripe.android.paymentsheet.state.PaymentElementLoader
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import com.stripe.android.R as PaymentsCoreR
+
+internal class GooglePayConfirmationDefinition(
+    private val googlePayPaymentMethodLauncherFactory: GooglePayPaymentMethodLauncherFactory,
+    private val userFacingLogger: UserFacingLogger?,
+) : ConfirmationDefinition<
+    GooglePayConfirmationOption,
+    ActivityResultLauncher<GooglePayPaymentMethodLauncherContractV2.Args>,
```

Comment:
Definitely as a follow up (if at all). I wonder if we can change this type from `ActivityResultLauncher<GooglePayPaymentMethodLauncherContractV2.Args>` to just `GooglePayPaymentMethodLauncherContractV2.Args` assuming all of these use `ActivityResultLauncher`.

Reply:
Yeah, thanks for entertaining my idea! Seems like the way it is is ideal!